pkgs/by-name/do/docker-init/package.nix

```diff
@@ -0,0 +1,47 @@
+{
+  stdenv,
+  fetchurl,
+  zstd,
+  lib,
+}:
+stdenv.mkDerivation rec {
```

Comment:
I think we should use the `finalAttrs` pattern here.

The `finalAttrs` pattern will let you remove the `rec` keyword ([see its implementation in Nix](https://github.com/NixOS/nix/blob/2d1d81114d72ace89ce08cd3bc93f4eb27a2975d/src/libexpr/nixexpr.cc#L121)).

Why is this pattern preferred to `rec` ?

Let's take this simple code example:

```nix
mkDerivation rec {
   foo = 1;
   bar = foo + 1;
}
```

and then `.overrideAttrs(old: { foo = 2; })`, you'll get `{ foo = 2; bar = 2; }` while with `finalAttrs` pattern, it would work correctly because it's a real fixed point.

Let me share a couple of useful links regarding the `finalAttrs` pattern:

1. History: https://discourse.nixos.org/t/avoid-rec-expresions-in-nixpkgs/8293
2. Documentation: https://nixos.org/manual/nixpkgs/unstable/#mkderivation-recursive-attributes
3. Recent example of implementation: https://github.com/NixOS/nixpkgs/compare/17f96f7b978e61576cfe16136eb418f74fab9952..9e6ea843e473d34d4f379b8b0d8ef0425a06defe

Feel free to reach out if you need some assistance.

Reply:
Hey, that seems like a good idea, I'll get on that.

Reply:
I have a question though, unrelated. In this derivation we both have the `version` field and the `tag` field. Only the `tag` field needs to get updated. Couldn't it cause confusion if someone tried to say, manually update the version used by overriding the package and only changing `version` to something else? They wouldn't see any changes since `tag` actually takes care of that stuff. So my question is, should I change `tag` to `version` instead perhaps and just remove the human-readable version number?

Something like:
```diff
-version = "v1.30.0";
+version = "157355";
-tag = "157355";
```